### PR TITLE
GUACAMOLE-1643: Document KSM user, connection-group level configuration.

### PR DESCRIPTION
**Pull request by jmuehlner**

### CODE REVIEW

File: src/vault.md [hunk 1]
```diff
@@ -170,6 +174,28 @@ All other configuration properties are optional.
...
+In addition to the required global `ksm-config` configuration blob, Guacamole
+can also be configured with user or connection group KSM configuration, which
+will pull additional secrets _only_ when not already available. If a secret can
+be pulled using the `ksm-config` global KSM config, it will always be used.
+Failing that, if a secret is available using the connection grop config, that
+value will be used. Only when neither the global or containing connection group
+KSM configs define a secret will the user KSM config be used. Note also that
+user KSM configs will be disabled unless ``ksm-allow-user-config` is set to true.
+
+These KSM config values can be set directly in the webapp, on the connection
```
Comment (jmuehlner) [inline src/vault.md:194]: Should I include screenshots of the edit pages for these? I could see them being useful, but it'd also be pretty cluttered if we included screenshots like this all the time. The screenshots would look something like the following:
image
image

Comment (mike-jumper) [reply src/vault.md:194]: I think things are clear without screenshots, but we should probably add links to the documentation covering the "Preferences" screen and the connection group editor.

**Comment by necouchman** [comment]
@jmuehlner Just FYI - both your commit message and your pull request have `GUACAMOLe` instead of `GUACAMOLE` (lower e).

**Comment by jmuehlner** [comment]
> @jmuehlner Just FYI - both your commit message and your pull request have `GUACAMOLe` instead of `GUACAMOLE` (lower e).

Oops! Fixed.

File: src/vault.md (hunk 1 above)
Comment (jmuehlner) [reply src/vault.md]: Good call - linked.

**Comment by jmuehlner** [comment]
Note: this PR also includes documentation updates for https://issues.apache.org/jira/browse/GUACAMOLE-1656 (per-user vault configuration). It probably doesn't make sense to try to seperate these changes, since they're so interdependent with the other KSM changes - should I update the commit message to also include `GUACAMOLE-1656`?

**Comment by mike-jumper** [comment]
> Note: this PR also includes documentation updates for GUACAMOLE-1656 and GUACAMOLE-1629. It probably doesn't make sense to try to seperate these changes, since they're so interdependent with the other KSM changes - should I update the commit message to also include `GUACAMOLE-1656`?

Sure - I think it'd be good to add a note to the relevant commit(s).

**Comment by jmuehlner** [comment]
> > Note: this PR also includes documentation updates for GUACAMOLE-1656 and GUACAMOLE-1629. It probably doesn't make sense to try to seperate these changes, since they're so interdependent with the other KSM changes - should I update the commit message to also include `GUACAMOLE-1656`?
> 
> Sure - I think it'd be good to add a note to the relevant commit(s).

Does this look reasonable?

**Comment by mike-jumper** [comment]
Looks reasonable to me.

File: src/vault.md [hunk 2]
```diff
@@ -115,6 +115,10 @@ question should be able to access.
 
    ![Application creation confirmation dialog showing the generated one-time token.](images/vault-ksm-004-generate-token.png)
 
+   This token can be used to generate a base-encoded configuration blob as
```
Comment (mike-jumper) [inline src/vault.md:118]: base64*-encoded?

Comment (jmuehlner) [reply src/vault.md]: Good catch - fixed.

**Review by mike-jumper — APPROVED** [review]

**Pull request merged**